v6.2.0 (#427)
MetaMask/controllers: CHANGELOG.md

## [Unreleased]

## [6.2.0] - 2021-03-23

### Added

> - Adds the Notification Controller (to support "what's new" type announcments in-app) ([#329](https://github.com/MetaMask/controllers/pull/329))

Reviewer: Nit: This should not be present tense, and there was a typo in "announcements"

```suggestion
- Add the Notification Controller (to support "what's new" type announcements in-app) ([#329](https://github.com/MetaMask/controllers/pull/329))
```

### Changed

- Explicitly add ethereumjs-tx as a package.json dependency ([#392](https://github.com/MetaMask/controllers/pull/392))
- Add `types` manifest field to package.json ([#391](https://github.com/MetaMask/controllers/pull/391))
- Use "options bag" for parameters for BaseControllerV2 constructor ([#388](https://github.com/MetaMask/controllers/pull/388))
- Ensure `uuid` dependency is type-checked ([#403](https://github.com/MetaMask/controllers/pull/403))
- Update TypeScript to v4.2 ([#369](https://github.com/MetaMask/controllers/pull/369))
- Asset metadata type conditionally requires error field, disallows for non-errors ([#395](https://github.com/MetaMask/controllers/pull/395))

> - Consistently use BN type for token balances ([#398](https://github.com/MetaMask/controllers/pull/398))

Reviewer: This should be categorized as a bug fix - I was able to find an example in mobile where the thing being fixed here would cause a crash.

Author: Done in 33436e3a

> - `approveTransaction` of TransactionController gets nonce from txMeta, falls back to network nonce ([#381](https://github.com/MetaMask/controllers/pull/381))

Reviewer: This doesn't seem like a good description :thinking: This commit added support for custom nonces. This should probably go under "Added" as well, since it's a new feature.

```suggestion
- Add support for specifying a custom nonce ([#381](https://github.com/MetaMask/controllers/pull/381))
```

Reviewer: I noticed that it's still under "Changed" - what do you think about the suggestion to recategorize it as 'Added"?

- Improve TransactionMeta type: `status` now an enum, error conditional on status, default error added for failed etherscan transaction ([#406](https://github.com/MetaMask/controllers/pull/406))
- `NetworkController` no longer a required controller of `TypedMessageManager` ([#416](https://github.com/MetaMask/controllers/pull/416))
- Update `selectedAddress` when identities are updated in `PreferencesController.updateIdentities` ([#415](https://github.com/MetaMask/controllers/pull/415))
- Add contract address validation to `AssetsContractController.getCollectibleTokenURI` ([#414](https://github.com/MetaMask/controllers/pull/414))

> - Add descriptive error messages to empty tothrow calls ([#422](https://github.com/MetaMask/controllers/pull/422))

Reviewer: Nit: We should use the correct capitalization for the method so that it's recognizable.

```suggestion
- Add descriptive error messages to empty `toThrow` call ([#422](https://github.com/MetaMask/controllers/pull/422))
```
